docs/GetStarted/generic_vulkan_env_setup.md

```diff
@@ -95,21 +95,61 @@ $ LD_LIBRARY_PATH={PATH_TO_VULKAN_SDK}/x86_64/lib/
 $ LD_PRELOAD=libvulkan.so.1
 ```
 
+## Vulkan debugging and profiling
+
+### RenderDoc
+
+RenderDoc is an awesome tool that one can use to capture and introspect Vulkan
+applications. It can be downloaded from https://renderdoc.org/ or compiled
+from [source][RenderDocSource].
+
 ## Vulkan development environment troubleshooting
 
 ### Useful environment variables
 
 There are a few environmet variables that can alter the default Vulkan loader
 behavior and print verbose information, notably:
 
-*   `VK_LOADER_DEBUG`: enable loader debug messages.
-*   `VK_ICD_FILENAMES`: force the loader to use a specific ICD.
-*   `VK_INSTANCE_LAYERS`: force the loader to enable the given layers.
+*   `VK_LOADER_DEBUG`: enable loader debug messages. Setting it to `all` will
+    enable the most verbose logging from the loader. This is especially useful
+    when trying to see what layers/ICDs are searched and used.
+*   `VK_ICD_FILENAMES`: force the loader to use a specific ICD. This is
+    especially useful when you have multiple Vulkan capable devices and want
+    to select which one to use manually.
+*   `VK_INSTANCE_LAYERS`: force the loader to enable the given layers. For
+    example, You can force enable `VK_LAYER_LUNARG_api_dump` to have a detailed
+    dump of all Vulkan API calls made by the application. You can force enable
+    `VK_LAYER_LUNARG_core_validation` to validate the API calls made by the
+    application.
 *   `VK_LAYER_PATH`: override the loader's standard layer libary search folders.
 
 Please see the [Vulkan loader's documentation][VulkanLoaderEnvVars] for detailed
 explanation for these variables.
 
+### Setting environmet variables for Bazel test
+
+Bazel runs tests in a sandbox and environment variables must be passed through
+to the test runner. Consider putting environment setup in a `user.bazelrc` to
+trouble shooting a test run by Bazel. For example:
```

Comment:
```suggestion
to the test runner. Consider putting environment setup in a `user.bazelrc`. For
example:
```

Keep the original language or adjust. When troubleshooting, you would just set the arguments directly on the command line, you persist common configurations to bazelrc files to save typing.

Developers are also free to add their own configurations, like

```
test:vkswiftshader --test_env="LD_LIBRARY_PATH=...
test:vkswiftshader --test_env="VK_LAYER_PATH=...
test:vknative --test_env="LD_LIBRARY_PATH=...
test:vknative --test_env="VK_LAYER_PATH=...
```

then use with `bazel test --config=vkswiftshader ...`

(actually now that I'm typing that out I might do that setup on my Linux machine... I've been saving bash history that sets all the flags :P)

Reply:
Good points! I've updated suitable places to have this handy settings.